wp-admin/js/widgets/media-gallery-widget.js

```diff
@@ -0,0 +1,297 @@
+/* eslint consistent-this: [ "error", "control" ] */
+(function( component, $ ) {
+	'use strict';
+
+	var GalleryWidgetModel, GalleryWidgetControl, GalleryDetailsMediaFrame;
+
+	/**
+	 * Custom gallery details frame.
+	 *
+	 * @class GalleryDetailsMediaFrame
+	 * @constructor
+	 */
+	GalleryDetailsMediaFrame = wp.media.view.MediaFrame.Post.extend( {
+
+		/**
+		 * Create the default states.
+		 *
+		 * @returns {void}
+		 */
+		createStates: function createStates() {
+			this.states.add([
+				new wp.media.controller.Library({
+					id:         'gallery',
+					title:      wp.media.view.l10n.createGalleryTitle,
+					priority:   40,
+					toolbar:    'main-gallery',
+					filterable: 'uploaded',
+					multiple:   'add',
+					editable:   true,
+
+					library:  wp.media.query( _.defaults({
+						type: 'image'
+					}, this.options.library ) )
+				}),
+
+				// Gallery states.
+				new wp.media.controller.GalleryEdit({
+					library: this.options.selection,
+					editing: this.options.editing,
+					menu:    'gallery'
+				}),
+
+				new wp.media.controller.GalleryAdd()
+			]);
+		}
+	} );
+
+	/**
+	 * Gallery widget model.
+	 *
+	 * See WP_Widget_Gallery::enqueue_admin_scripts() for amending prototype from PHP exports.
+	 *
+	 * @class GalleryWidgetModel
+	 * @constructor
+	 */
+	GalleryWidgetModel = component.MediaWidgetModel.extend( {
+		/**
+		 * Remove an attachment ID from attachments.
+		 *
+		 * @param {Integer} id - Attachment id to remove from attachments.
+		 * @returns {void}
+		 */
+		removeAttachmentId: function removeAttachmentId( id ) {
+			var attachments, newAttachments;
+			attachments = JSON.parse( this.get( 'attachments' ) );
+			newAttachments = _.filter( attachments, function( attachment ) {
+				return attachment.id !== id;
+			} );
+			this.set( {
+				'attachments': JSON.stringify( newAttachments ),
+				'ids': _.map( newAttachments, 'id' )
+			} );
+		}
+	} );
+
+	/**
+	 * Gallery widget control.
+	 *
+	 * See WP_Widget_Gallery::enqueue_admin_scripts() for amending prototype from PHP exports.
+	 *
+	 * @class GalleryWidgetControl
+	 * @constructor
+	 */
+	GalleryWidgetControl = component.MediaWidgetControl.extend( {
+		/**
+		 * Render preview.
+		 *
+		 * @returns {void}
+		 */
+		renderPreview: function renderPreview() {
+			var control = this, previewContainer, previewTemplate, attachments;
+
+			previewContainer = control.$el.find( '.media-widget-preview' );
+			previewTemplate = wp.template( 'wp-media-widget-gallery-preview' );
+
+			if ( control.model.get( 'ids' ).length && ! control.model.get( 'attachemnts' ) ) {
```

Comment:
`! control.model.get( 'attachemnts' )` this looks like it can be eliminated. At least, it is misspelled.